packages/transaction-controller/src/gas-flows/OracleLayer1GasFeeFlow.ts

```diff
@@ -103,7 +102,7 @@ export abstract class OracleLayer1GasFeeFlow implements Layer1GasFeeFlow {
     transactionMeta: TransactionMeta,
   ): TransactionMeta['txParams'] {
     return {
-      ...omit(transactionMeta.txParams, 'gas'),
+      ...transactionMeta.txParams,
       gasLimit: transactionMeta.txParams.gas,
```

Comment:
So this is an interesting one.

We're generating params here to pass to `prepareTransaction` which ultimately uses `TransactionFactory.fromTxData` from `@ethereumjs/tx` and that doesn't appear to have `gas` on the `TypedTxData` type. Though we also use this for the main gas estimation in `addTransaction`.

Which makes me wonder how the `gas` property is being processed currently?

Can we use `gas` and decode the transaction sent to `eth_sendRawTransaction` to verify it's being included correctly? Assuming the UI isn't setting it somewhere also.

Reply:
In both cases where the dApp request includes specific `gas` and `gasLimit` properties, the decoded raw tx includes a `gasLimit` property, and not a `gas` one

Reply:
It doesn't exactly match it: if the request includes gas OR gasLimit, the txParams include both gas AND gasLimit, but the decoded raw transaction ONLY includes gasLimit.

Reply:
Apologies, I meant match it in that the dApp gas limit is kept, regardless of which property is used.

Reply:
Ah yes, the decoded raw tx includes gasLimit when the request includes either gas or gasLimit (or even none of the two, in which case gas is estimated)